### ISSUE TITLE
Default comment type

### ISSUE THREAD

**Issue by jdanish** [netcreateorg/netcreate-itest#244]
Currently the system crashes if there isn't a comment with the slug "cmt".  We should allow setting the default or just use the top comment type in the list.  Low priority since we can just use cmt for now ...

**Comment by benloh** [comment]
# TO DO

- [X] ~`template-schema.js` needs to define `MOD.TEMPLATE`~ `template-schema` is only used for JSON editor, which has been depreated.
- [X] Make sure `templateEditor-mgr` gracefully handles `COMMENTTYPES` -- uses the `TEMPLATE` object whole, so `commentTypes` should be included.
- [X] Make sure `importexport-mgr` gracefully handles `COMMENTTYPES` -- only handles Node and Edge import/export, template is not used.
- [x] Make sure `_default.template.toml` has a `COMMENTTYPES` entry
- [x] Make sure importing `COMMENTTYPES` works.
- [X] Make sure exporting `COMMENTTYPES` works.
- [x] Check fallbacks
   - [x] No comments defined in template, and no DEFAULT_CommentTypes defined => fall back to DEFAULT_COMMENTTYP
   - [x] No comments defined in template with DEFAULT_CommentTypes defined => fall back to DEFAULT_CommentTypes
   - [x] Comment type defined in template => Use template
 - [x] Check defaults
    - [x] No comment slug `cmt` -- make sure existing comments falls back to valid comment type
    - [x] First slug in types used

**Comment by benloh** [comment]
@jdanish In establishing default comment types, here's what I have in mind.  The idea is to provide multiple levels of fallback so that we are always guaranteed to have a comment type available:

1. You can define the comment types in your project template file directly by editing your `<projectname>.template.toml` file.
2. If you create a new project from the `_default.template.toml` file, the default template will define a generic comment type for you.  You can then use this as a template for other projects.  What should they be?  `cmt` and `tellmemore`?  This will be the default comment types whenever you create a new project.  You can update `_default.template.toml` to create a new set of comment types if you want to different starting set.  This allows you to set the comment types without changing the code.  e.g. for a given server, you can define a set of comment types for all the graphs that do not have a comment type.
```
{
  slug: 'cmt',
  label: 'Comment', // comment type label
  prompts: [
    {
      format: 'text',
      prompt: 'Comment', // prompt label
      help: 'Use this for any general comment.',
      feedback: ''
    }
  ]
},
 {
    id: 'tellmemore',
    label: 'Tell me more', // comment type label
    prompts: [
      {
        prompt: 'Please tell me more', // prompt label
        help: 'Can you tell me more about ... ',
        feedback: ''
      }
    ]
  },
```
3. If a comment type is not defined in the project template, fall back to using a set of two comments as "burned" into the `dc-comments` code.  So if you mess up the project template, or revert to an old template, or remove the comment types defined in `_default.template.toml` you can still rely on the code to provide some basic types.  What should they be?  Same set of `cmt` and `tellmemore`?   If you want to spin up or clone multiple servers from a common code base, you would modify the `DEFAULT_CommentTypes` constant in the code.
```
{
  slug: 'cmt',
  label: 'Comment', // comment type label
  prompts: [
    {
      format: 'text',
      prompt: 'Comment', // prompt label
      help: 'Use this for any general comment.',
      feedback: ''
    }
  ]
},
 {
    id: 'tellmemore',
    label: 'Tell me more', // comment type label
    prompts: [
      {
        prompt: 'Please tell me more', // prompt label
        help: 'Can you tell me more about ... ',
        feedback: ''
      }
    ]
  },
```
or perhaps a generic `evidence` type?
```
  {
    slug: 'evidence',
    label: 'Evidence Critique or Suggestion',
    prompts: [
      {
        format: 'dropdown',
        prompt: 'Is this supported by evidence?', // prompt label
        options: ['😀 Yes', '🤔 Some', '🥲 No'],
        help: 'Select one.'
      },
      {
        format: 'text',
        prompt: 'What would you change?', // prompt label
        help: 'Please be specific to help your friend.'
      }
    ]
  },
```
4. Finally, if the `DEFAULT_CommentTypes` is not defined, e.g. you really don't to have multiple comment types, the system will back to the base comment definition.  This way you can easily reset to a base comment type.
```
{
  slug: 'cmt',
  label: 'Comment', // comment type label
  prompts: [
    {
      format: 'text',
      prompt: 'Comment', // prompt label
      help: 'Use this for any general comment.',
      feedback: ''
    }
  ]
},
```

Does that work for you?  Do you want to "burn" in a different set?

**Comment by jdanish** [comment]
Yes, this sounds great with cmt and tellmemore being the usual defaults, cmt being the backup if all else fails, and everything being editable via the template. I think cmt and tellmemore are the broadest, and something like evidence can be added project by project. Thanks!

**Comment by benloh** [comment]
Should be fixed with e3893fe55fb25d1fc885ba8004a6625af3e2e062
